src/dune_engine/sandbox.ml

```diff
@@ -90,15 +103,22 @@ let link_function ~(mode : Sandbox_mode.some) =
         | false -> fun src dst -> Io.portable_symlink ~src ~dst)
      | Copy ->
        fun src dst ->
-         let { Unix.st_kind; _ } = Path.Untracked.stat_exn src in
-         copy_recursively st_kind ~src ~dst
+         let what =
+           let { Unix.st_kind; st_perm; _ } = Path.Untracked.stat_exn src in
+           match st_kind with
+           | S_DIR -> `Dir st_perm
```

Comment:
Let's create a regular type for this variant and put it in a module with something like `of_unix_stats` to avoid duplicating this code.

Reply:
I could do a regular variant, but it actually won't prevent duplicating the code. That's because:

1. At this call site, we get the file kind from `Unix.stat`
2. In the other call sites, we get the file kind from `Path.readdir_with_kinds`

Would you still like the regular variant?

Reply:
Yes, I think so. Converting stats around is boring code, and it's best to keep it in a little helper module.